### ISSUE TITLE
doc requirements in [tool.poetry.dev-dependencies]?

### ISSUE THREAD

**Issue by Jacob-Stevens-Haas** [andgoldschmidt/derivative#11]
Hey Andy, :wave:, I'm starting to work on the Kalman stuff and trying to make sure the docs build correctly.  Apparently, poetry automatically installs the dev dependencies but not optional dependencies which must be [specified individually](https://python-poetry.org/docs/pyproject/#extras).  Can I copy all the documentation optional dependencies into the `[tool.poetry.dev-dependencies]` section of pyroject.toml?

Related: I can't build the docs even with the optional dependencies: mention this in #12. 

Also, I have no idea why my /env/ folder isn't showing up since there's no .gitignore to ignore it, but maybe some poetry magic?

**Comment by andgoldschmidt** [comment]
Hi! Yes, I've specified everything for the docs under [tool.poetry.extras] so it can be installed via
`poetry install -E docs`

As to "Pandoc wasn't found" in #12 ... it seems like this should be a dependency, see https://nbsphinx.readthedocs.io/en/0.2.15/installation.html#pandoc

Did you test to see if installing pandoc in a clean poetry venv after installing derivative was enough to let you build the docs?

**Comment by andgoldschmidt** [comment]
I tested and was able to build the documentation within a fresh poetry venv without adding pandoc, so I'll go ahead and make the edits necessary for merging #12 under the current install requirements.

**Comment by Jacob-Stevens-Haas** [comment]
Thanks for adding the docs dependencies :partying_face: 

> Did you test to see if installing pandoc in a clean poetry venv after installing derivative was enough to let you build the docs?
No, it still didn't build.  Here's the complete issue: 
<details>
<summary> DeadKernelError </summary>

```
(env) xenophon@pontus:~/github/derivative/docs$ make html
Running Sphinx v4.5.0
WARNING: html_static_path entry '_static' does not exist
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 159 source files that are out of date
updating environment: [new config] 159 added, 0 changed, 0 removed
/home/xenophon/github/derivative/docs/env/lib/python3.8/site-packages/matplotlib/animation.py:887: UserWarning: Animation was deleted without rendering anything. This is most likely not intended. To prevent deletion, assign the Animation to a variable, e.g. `anim`, that exists until you have outputted the Animation using `plt.show()` or `anim.save()`.
  warnings.warn(
/home/xenophon/github/derivative/docs/env/lib/python3.8/site-packages/lockfile-0.12.2.dist-info/DESCRIPTION.rst:1: ERROR: Unknown target name: "fasteners".                                                
/home/xenophon/github/derivative/docs/env/lib/python3.8/site-packages/lockfile-0.12.2.dist-info/DESCRIPTION.rst:1: ERROR: Unknown target name: "oslo.concurrency".
/home/xenophon/github/derivative/docs/env/lib/python3.8/site-packages/lockfile-0.12.2.dist-info/DESCRIPTION.rst:1: ERROR: Unknown target name: "openstack-dev".

Notebook error:
DeadKernelError in env/lib/python3.8/site-packages/nbclient/tests/files/Autokill.ipynb:
Kernel died
make: *** [Makefile:20: html] Error 2
```
</details>

Also, whether running `poetry install` or `poetry install -E docs` I get the warning:
```
Warning: The lock file is not up to date with the latest changes in pyproject.toml. You may be getting outdated dependencies. Run update to update them.
```

**Comment by andgoldschmidt** [comment]
Ok, so installing pandoc via poetry is not enough, see: https://stackoverflow.com/a/71585691

The docs built fine on osx without pandoc, but not on linux. I've added pandoc to the poetry toml as part of the docs, but users must also get pandoc via apt-get because the python package only acts as a wrapper for this executable.

I also took care of that `WARNING: html_static_path entry '_static' does not exist`, we don't have any _static files to bother with.

---

Regarding the 2nd warning, make sure to interact with poetry packages in a poetry shell, and update / install:
`poetry shell`
`poetry update`
`poetry install -E docs`

**Comment by Jacob-Stevens-Haas** [comment]
Hm, I made sure to get the pandoc binaries via `dpkg` on my system and then installed pandoc via poetry.  I called `make html` in the poetry shell.  I'm right now trying to try to see if they linked correctly.

Also, the matrix division A \ b in the kalman method might have been causing the Dead Kernel error if memory was an issue?  I'll try building the docs on the old version of `derivative`, and if that works, trying `scipy.sparse.linalg.spinv` instead of converting to dense matrices and what I'm currently using, `numpy.linalg.inv`.
